web/satellite/src/store/modules/credits.ts

```diff
@@ -0,0 +1,46 @@
+// Copyright (C) 2019 Storj Labs, Inc.
+// See LICENSE for copying information.
+
+import { CREDIT_USAGE_ACTIONS } from '@/utils/constants/actionNames';
+import { StoreModule } from '@/store';
+import { CREDIT_USAGE_MUTATIONS } from '@/store/mutationConstants';
+import { CreditsApi, CreditUsage } from '@/types/credits';
+
+const { FETCH } = CREDIT_USAGE_ACTIONS;
+const { SET, CLEAR } = CREDIT_USAGE_MUTATIONS;
+
+export function makeCreditsModule(api: CreditsApi): StoreModule<CreditUsage> {
+    return {
+        state: new CreditUsage(),
+
+        mutations: {
+            [SET](state: CreditUsage, creditUsage: CreditUsage) {
+                state.availableCredits = creditUsage.availableCredits;
+                state.usedCredits = creditUsage.usedCredits;
+                state.referred = creditUsage.referred;
+            },
+            [CLEAR](state: CreditUsage): void {
```

Comment:
In order to avoid naming collision, maybe we can change `clear` in mutation to be `reset_state` since mutation meant to change state and `clear` is more like an action?

Reply:
yeah, smth like that, but i'll do it in another pr